# 인사

안녕하세요 ~~ 25기 여러분 반갑습니다 ~~  

드디어 정규세션이 시작되었네요.  

뭘 과제로 내볼까 고민하다가, 요즘 데이터 분석 채용 공고 자격 요건에 필수로 들어있는.... A/B TEST가 좋겠다!!고 생각했습니다.  

데이터 분석쪽이 아니더라도, 다양한 도메인에서 쓸 수 있는 방법이라고 생각해요.


A/B TEST의 개념부터 기초적인 과정을 이해해볼 수 있게 내봤어요. 근데 설명해야할 게 많다보니 텍스트가 너무 많아졌네요.

읽기 힘들어도 ㅎㅎ... 열심히 해주시고, 질문 사항은 저에게 연락 남겨주세요.  

참고로, 정규세션 발표 외의 내용을 구글링하고 공부해야만 풀 수 있는 문제들이 포함되어 있습니다.

제 이름... 아시죠...? 24기 박현서입니다....



<br>

# Q1. A/B TEST란 무엇일까요?   
또한, A/B TEST시 유의할 사항들을 초두효과를 포함해 기술해주세요.

정규세션에서 발표한 내용 외에 추가적으로 공부한 개념을 작성하셔도 됩니다.
정의와 함께 예시 하나만 들어주세요 ~~

## A1. 답안 작성  
A/B TEST란, 두 가지의 대안을 실제 사용자에게 무작위로 노출해 어떤 안이 더 좋은 성과를 내는지 통계적으로 검증하는 실험 방법이다.
A/B TEST 진행 시 대표적으로 유의할 점은 4가지이다.
1. 명확한 가설과 단일 변경: 하나의 요소만 타겟으로 삼고, 명확히 세운 가설을 검정해야 정확한 효과를 확인할 수 있다.
2. 충분한 수의 표본을 무작위 배정: 사용자 특성(연령, 지역, 기기 등)이 A/B에 골고루 분산돼야 한다.
3. 충분한 실험 기간 : 사용자에게 나타날 수 있는 초두효과(처음 접한 정보나 디자인에 더 큰 영향을 받는 현상)를 견제하기 위해, 초기 반응이 아니라 안정화된 행동을 관찰할 수 있는 충분한 기간이 필요하다.
4. 외부 요인 통제 : 실험 중 이벤트가 발생해 영향을 받는 걸 조심하고, 가능하면 동시에 실험하는 게 좋다.





---



 A/B TEST를 하기 위해서는 1-pager를 작성해야합니다.  

1-pager란 문제 정의, 관찰하고자하는 실험대상, 가설설정, 평가지표 등을 작성하는  
간단히 말해 계획서를 의미하는데요,

다른 부서와의 소통을 위한 역할 또,과거에 어떤 테스트를
어떻게 진행했는지에 대한 기록의 역할을 하기도 합니다.


 A/B TEST의 1-pager는 복잡한 실험 과정을 한눈에 파악하고, 의사결정권자가 "그래서 이 실험을 왜 했고, 결과가 무엇이며, 무엇을 할 것인가?"를 바로 이해할 수 있도록 작성하는 것이 핵심입니다.

예시를 통해 어떤 항목이 있고, 각 항목에 어떤 내용을 채우는 건지 보여드릴게요.



---



**예시**   
(푸는 거 아니고, 예시입니다! 읽으면서 이런 부분은 이런 내용을 채우는 거군... 이해하시면 돼요!)  




 📊 [ThinkGood] 메인 홈 검색 모듈 상위 노출 실험 1-Pager

---

**1. 실험 전략 및 배경 (Strategy & Observation)**

| 항목 | 내용 |
| :--- | :--- |
| **Target User** | 특정 공모전을 찾으려는 목적이 뚜렷하지만, 현재 하단에 위치한 검색바를 즉각적으로 발견하지 못한 유저 |
| **Observation** | **[데이터 분석 결과]** <br> • 검색 이용 유저의 상세 진입 전환율(CVR)은 40%, 미이용 유저의 전환율은 25%임 <br> • 현재 검색 이용률은 30%로, 검색 가시성 개선을 통한 전환율 상승 여력이 큼 |
| **Problem Statement** | 검색바가 배너 하단에 위치하여 시인성이 낮으며, 유저가 정보를 탐색하기 전 이탈할 확률이 높음 |
| **Hypothesis** | 하단 검색바를 페이지 **최상단으로 노출**시켜 검색 접근성을 높인다면, 전체 공모전 상세 진입률(CVR)이 개선될 것이다 |

---

 **2. 실험군 설계 (Experimental Groups)**

<table>
  <tr>
    <td align="center" width="50%">
      <b>A (Control) - 기존안</b><br>
      <img src="https://i.postimg.cc/kgcLc7Qy/Gemini-Generated-Image-enqvuhenqvuhenqv-(1).png" width="100%"><br>
      <sub>검색바 하단 위치 (기존 UI)</sub>
    </td>
    <td align="center" width="50%">
      <b>B (Variant) - 실험안</b><br>
      <img src="https://i.postimg.cc/0N51Rz5M/Gemini-Generated-Image-enqvuhenqvuhenqv.png" width="100%"><br>
      <sub>검색바 최상단 배치 (변경 UI)</sub>
    </td>
  </tr>
</table>

---

 **3. 지표 및 통계 설계 (Metrics & Statistics)**

> **핵심 측정 지표**
> * **Primary Metric (주지표):** 공모전 상세 페이지 진입률 (Detail Page CVR)
> * **Secondary Metric (보조지표):** 검색바 클릭률 (Search CTR)
> * **Guardrail Metric (방어지표):** 메인 페이지 이탈률 (Bounce Rate)

| 통계적 설계 항목 | 설정값 | 비고 |
| :--- | :--- | :--- |
| **Significance Level ($\alpha$)** | 5% | 유의수준 0.05 |
| **Statistical Power ($1-\beta$)** | 80% | 검정력 유지 |
| **Minimum Sample Size** | **총 92,000명** | 각 그룹당 46,000명 할당 |
| **Required Period** | **최소 4일** |  

---

**4. 위험 관리 및 중단 기준 (Risk & Andon)**

* **Trade-Off:**   
    검색바 상단 노출로 인해 메인 배너의 시각적 주목도가 분산될 수 있으며, 기존 UI에 익숙한 유저에게 일시적인 초두 효과(Primacy Effect)에 의한 혼란이 발생할 수 있음.
* **Andon (중단 기준):**
    1. 대조군 대비 전체 상세 진입률이 **5% 이상 유의미하게 하락**할 경우
    2. UI 변경으로 인한 페이지 오류나 유저 이탈률이 **3일 연속 급증**할 경우

---







<br>

# Q2. 1-pager를 작성해봅시다. (꼬리문제 3 + 보고서 채우는 것1)
저희가 만든 데모 사이트 링크를 첨부해드릴테니, 마음껏 보신 뒤에 위의 예시처럼 1-pager를 작성해보기 바랍니다.

여기서 실험군 설계에 들어가는 사진은 굳이 넣을 필욘 없고 말로만 설명해주셔도 됩니다. 아니면 gemini에게 요청해볼 수도 있겠네요!  

또한 문제탐색에 필요한 cvr과 같은 지표들은 직접 가정하고 하시면 됩니다. 다만, 해당 지표들이 큰 문제가 있는 지표여야... a/b test를 진행하는 이유가 되겠죠!?


정리해보면, 저희 웹사이트 보시고 '이건 여기에 배치하면 더 좋을 거 같은데?' 하는 것들을 찾아서 모든 것을 가정한 뒤에 쓰시면 됩니다!

[데모 웹사이트](https://huggingface.co/spaces/HanJaeHui/ThinkGood_Recommendation_System)


추가적으로 아래 실험군 항목에 사진을 넣으실 분들은 제가 아래 첨부드리는 사이트에 방문하셔서 해당 이미지를 코랩에 넣을 수 있는 형태로 변환한 링크를 복붙하시면 됩니다.  

(.png로 끝나는 링크 복붙하시면 됩니다.)

[포스트 이미지 사이트](https://postimages.org/)



<br>

## Q2-1. 실험군을 설계 해봅시다.
아래 보고서를 채워 주세요.

## A2-1. 아래 보고서 항목 설명을 참고하여 '[ ]' 부분을 지우고 채워주세요.  
  



📊 [프로젝트명] 실험 기획안 1-Pager

---

1. 실험 전략 및 배경 (Strategy & Observation)

| 항목 | 내용 |
| :--- | :--- |
| **Target User** | 찾으려는 공모전 관심 분야가 명확하지만, 관심 분야 선택지 탭에서 본인의 관심 분야 항목을 쉽게 발견하지 못한 유저 |
| **Observation** | **데이터 근거** <br> • 관심분야 탭 이용 유저의 상세 페이지 진입률은 40%, 미이용 유저의 진입률은 25% <br> • 현재 관심분야 탭 이용률은 30%로, 관심분야 탭 가시성 개선을 통한 전환율 상승 여력이 큼 |
| **Problem Statement** | 관심분야 탭이 좌측에 작게 배치되어 시인성이 낮으며, 유저가 관심분야의 공모전 정보를 탐색하기 전 이탈할 확률이 높음 |
| **Hypothesis** | **좌측 관심 분야 탭을 가로로 긴 형태로 바꾼 뒤 검색바 상단으로 노출시킨다면, 접근성을 높아져 전체 공모전 상세 진입률(CVR)이 개선될 것이다.**  |

---

 2. 실험군 설계 (Experimental Groups)

<table>
  <tr>
    <td align="center" width="50%">
      <b>A (Control) - 기존안</b><br>
      <img src="[기존 시안 이미지 URL 입력]" width="100%"><br>
      <sub>[기존 UI에 대한 설명]</sub>
    </td>
    <td align="center" width="50%">
      <b>B (Variant) - 실험안</b><br>
      <img src="[변경 시안 이미지 URL 입력]" width="100%"><br>
      <sub>[변경 UI에 대한 설명]</sub>
    </td>
  </tr>
</table>


<br>

## Q2-2. 실험에 필요한 최소 test users 수를 구해봅시다.
유의수준 0.05, 검정력 80% 하에 수행해주세요.  


아래 공식 참고해서, 최소 test users 수를 구하는 코드 작성해주세요.

추가적으로 n은 산출하는데에 가정한 p1, p2도 같이 작성해주세요.

$$n = \frac{(Z_{\alpha/2} + Z_{\beta})^2 \cdot [p_1(1-p_1) + p_2(1-p_2)]}{(p_1 - p_2)^2}$$

## A2-2. 아래 질문에 대한 답과 코드를 채워주세요.

p1, p2는 각각 무엇을 의미하는 확률이고, 어떤 값으로 가정하셨나요?   
또한 위 공식을 보고 n을 구하기 위한 코드를 작성해주세요. 단, 라이브러리는 z값을 구하는 용도의 것만 사용해주세요.


### 답변:
p1 ​= A안 전체 CVR = 0.30⋅0.40+0.70⋅0.25 = 0.12+0.175 = 0.295

p2 = B안 전체 CVR = 0.50⋅0.40+0.50⋅0.25 = 0.20+0.125 = 0.325

In [ ]:
# A2-2 코드작성. 나온 n은 아래 Minimum Sample Size에 기재.

from scipy.stats import norm
import math

# =========================
# 1) 실험 조건
# =========================
alpha = 0.05      # 유의수준
power = 0.80      # 검정력(=1-β)

Z_alpha_over_2 = norm.ppf(1 - alpha/2)   # Z_{α/2} = Z_{0.975}
Z_beta = norm.ppf(power)                 # Z_{β} = Z_{0.8}

# =========================
# 2) p1, p2 가정 (문제 맥락 기반)
# =========================
p_tab = 0.40          # 탭 이용자 CVR
p_no_tab = 0.25       # 탭 미이용자 CVR

tab_rate_A = 0.30     # 현재 탭 이용률 (A안)
tab_rate_B = 0.50     # 개선 후 탭 이용률 가정 (B안)

p1 = tab_rate_A * p_tab + (1 - tab_rate_A) * p_no_tab
p2 = tab_rate_B * p_tab + (1 - tab_rate_B) * p_no_tab

# =========================
# 3) 표본수 계산 (그룹당 n)
# =========================
numerator = (Z_alpha_over_2 + Z_beta)**2 * (p1*(1-p1) + p2*(1-p2))
denominator = (p1 - p2)**2
n_per_group = numerator / denominator

print(f"Z_(alpha/2) = {Z_alpha_over_2:.4f}, Z_beta = {Z_beta:.4f}")
print(f"p1 = {p1:.4f}, p2 = {p2:.4f}, absolute lift = {p2-p1:.4f}")
print(f"Required n per group = {math.ceil(n_per_group)}")
print(f"Total users (A+B) = {math.ceil(2*n_per_group)}")


Z_(alpha/2) = 1.9600, Z_beta = 0.8416
p1 = 0.2950, p2 = 0.3250, absolute lift = 0.0300
Required n per group = 3727
Total users (A+B) = 7454


 Minimum Sample Size = 3727

## Q2-3. 실험에 필요한 최소 실험 기간을 구해봅시다.

이 실험에 필요한 최소 test users가 92000명이 나왔고  
해당 웹페이지 하루 평균 방문자수가 50000명 정도라고 할 때 우리는 최소 4일이 필요합니다.

비교하고 싶은 대상이 총 두개이고, 하나의 대상을 실험하기 위해선 최소 이틀은 필요하기 때문이죠!

위에서 구하신 test users수를 바탕으로, 하루 평균 방문자수를 가정한 뒤 기간을 설정해주세요.  
구하신 기간은 밑에 표에 기재해주시면 됩니다.



## A2-3. 아래 세질문에 답하신 뒤 아래 보고서를 채워주세요.

가정한 하루 평균 방문자 수?  2000명

산출된 최소 test users 수?   3727명
  
그래서 필요한 최소 기간은?   4일

## **아래 보고서도 채워주세요!**  
**'[ ]' 부분을 지우고 해당 내용에 부합하는 내용들을 채워주시면 됩니다.**

---

**3. 지표 및 통계 설계 (Metrics & Statistics)**

> **핵심 측정 지표**
> * **Primary Metric (주지표):** [실험의 성공 여부를 결정할 가장 중요한 지표]
> * **Secondary Metric (보조지표):** [함께 살펴볼 관련 지표]
> * **Guardrail Metric (방어지표):** [이 실험으로 인해 나빠지면 안 되는 지표]

| 통계적 설계 항목 | 설정값 | 비고 |
| :--- | :--- | :--- |
| **Significance Level ($\alpha$)** | 5% | 보통 5%(0.05) 설정 |
| **Statistical Power ($1-\beta$)** | 80% | 보통 80% 설정 |
| **Minimum Sample Size** | **총 3727명** | 그룹당 인원수 명시 |
| **Required Period** | **최소 4일** | 실험 진행 예정 기간 |

---

 **4. 위험 관리 및 중단 기준 (Risk & Andon)**

* **Trade-Off:**   
    관심 분야 탭을 검색바 상단에 노출시킴으로 인해 검색바에 대한 시각적 집중도가 분산될 수 있으며, 기존 UI에 익숙한 유저에게 일시적인 초두 효과(Primacy Effect)에 의한 혼란이 발생할 수 있음.
* **Andon (중단 기준):**
    1. 대조군 대비 전체 상세 진입률이 **5% 이상 유의미하게 하락**할 경우
    2. UI 변경으로 인한 페이지 오류나 유저 이탈률이 **3일 연속 급증**할 경우

---

<br>

# Q3. A/B TEST를 수행해봅시다.

1-pager를 작성했으면 직접 test도 수행해봐야겠죠!  
위에 제가 만들었던 1-pager를 바탕으로 검정을 해볼게요.  
대표적인 검정방법에는 t-test가 있고 카이제곱 겁정이 있습니다.  


---



[실험 가설]  
귀무가설 ($H_0$): 메인 홈 검색 모듈의 위치(하단 vs 최상단)와 상세 페이지 진입 여부 사이에는 상관관계가 없다. 즉, 두 그룹의 상세 페이지 진입률($CVR$)은 동일하다.$$H_0: p_A = p_B$$대립가설 ($H_1$): 메인 홈 검색 모듈을 최상단에 노출하는 것은 상세 페이지 진입 여부에 영향을 미친다. 즉, 실험군(B)의 상세 페이지 진입률이 대조군(A)보다 높을 것이다.$$H_1: p_A < p_B$$


<br>

## Q3-1. 위 가설에 적합한 검정 이름과 왜 해당 검정을 써야하는지 이유도 작성해주세요. (t test와 카이제곱검정 중)

## A3-1.  검정 이름과 이유를 설명해주세요.

답변: 위 가설에 필요한 검정은 카이제곱 검정이고,  

이유 : 본 실험의 반응 변수는 상세 페이지 진입 여부로 이진형 변수이며, 관심 대상은 평균이 아닌 두 집단 간 진입률(CVR)의 차이다. 따라서 두 범주형 변수의 독립성을 검정하는 카이제곱 검정이 더 적절하다.

<br>

In [ ]:
'''''''''''''''''''''''''''''''''''''''
제가 세운 A/B TEST 설계 실험 코드입니다.
'''''''''''''''''''''''''''''''''''''''

import pandas as pd
from scipy.stats import chi2_contingency

# 1. 관측 데이터 입력 (실제 실험 결과 수치로 변경하세요)
# [상세진입 성공 수, 상세진입 실패 수]
control_results = [11500, 34500]  # 대조군(A)
variant_results = [12190, 33810]  # 실험군(B)

# 2. 분할표(Contingency Table) 생성
data = [control_results, variant_results]

# 3. 카이제곱 검정 수행
chi2, p_value, dof, expected = chi2_contingency(data)

# 4. 결과 출력
print(f"카이제곱 통계량: {chi2:.4f}")
print(f"P-value: {p_value:.10f}")

alpha = 0.05
if p_value < alpha:
    print(f"결과: P-value가 {alpha}보다 작으므로 귀무가설을 기각합니다.")
    print("=> 검색바 최상단 노출이 상세 페이지 진입률에 유의미한 변화를 주었습니다.")
else:
    print(f"결과: P-value가 {alpha}보다 크므로 귀무가설을 기각할 수 없습니다.")
    print("=> 검색바 위치 변경의 효과가 통계적으로 증명되지 않았습니다.")

카이제곱 통계량: 26.9884
P-value: 0.0000002047
결과: P-value가 0.05보다 작으므로 귀무가설을 기각합니다.
=> 검색바 최상단 노출이 상세 페이지 진입률에 유의미한 변화를 주었습니다.


실제 데이터가 없어서, 저도 필요한 데이터는 가정했답니다.  

이제 25기 분들의 A/B TEST를 수행해보시죠!

<br>

## Q3-2. Q2에서 작성한 실험 수행을 위해 가설을 설정해주세요.


## A3-2. 아래 두 가설 내용을 채워주세요.
  
[실험 가설]  
귀무가설 ($H_0$): 메인 홈 관심 분야 모듈의 위치(좌측 vs 상단)와 상세 페이지 진입 여부 사이에는 상관관계가 없다. 즉, 두 그룹의 상세 페이지 진입률($CVR$)은 동일하다.$$H_0: p_A = p_B$$대립가설 ($H_1$): 메인 홈 관심 분야 모듈을 검색바 상단에 노출하는 것은 상세 페이지 진입 여부에 영향을 미친다. 즉, 실험군(B)의 상세 페이지 진입률이 대조군(A)보다 높을 것이다.$$H_1: p_A < p_B$$


<br>

## Q3-3. 위 가설을 수행하기 위해 어떤 검정 방법을 사용해야하고 이유는 무엇인지 작성해주세요.

## A3-3. 아래 검정 이름과 이유를 채워주세요.
답변: 위 가설에 필요한 검정은 카이제곱 검정이고,  

이유 : 본 실험의 반응 변수는 상세 페이지 진입 여부로 이진형 변수이며, 관심 대상은 평균이 아닌 두 집단 간 진입률(CVR)의 차이다. 따라서 두 범주형 변수의 독립성을 검정하는 카이제곱 검정이 더 적절하다.

<br>

## Q3-4. 위에서 답한 것을 토대로, 검정을 수행해주세요. (t test와 카이제곱 검정 중)


## A3-4 아래 셀에 코드 작성해주세요. 그리고!!!
그리고 나온 값을 통해 수행하신 a/b test 결과를 설정하신 가설에 맞게 해석해주세요.

In [1]:
### A3-4. 코드작성

import numpy as np
from scipy.stats import chi2_contingency

# =========================
# 1) 관측 데이터 입력
# =========================
# [상세진입 성공 수, 상세진입 실패 수]
control_results = [11500, 34500]  # 대조군(A)
variant_results = [12190, 33810]  # 실험군(B)

# 2x2 분할표
table = np.array([control_results, variant_results])

# =========================
# 2) 비율 계산 (방향성 확인용)
# =========================
a_success, a_fail = control_results
b_success, b_fail = variant_results

nA = a_success + a_fail
nB = b_success + b_fail

pA_hat = a_success / nA
pB_hat = b_success / nB
diff = pB_hat - pA_hat

print(f"pA_hat = {pA_hat:.6f}, pB_hat = {pB_hat:.6f}, (pB_hat - pA_hat) = {diff:.6f}")

# =========================
# 3) 카이제곱 검정 (양측 p-value)
# =========================
chi2, p_two, dof, expected = chi2_contingency(table, correction=False)

# =========================
# 4) 단측 가설(H1: pA < pB)로 p-value 변환
# =========================
# 방향이 H1과 같은지 확인 후 단측 p-value 구성
if pB_hat > pA_hat:
    p_one = p_two / 2
else:
    p_one = 1 - (p_two / 2)

print(f"Chi-square = {chi2:.4f}, p(two-sided) = {p_two:.10f}, p(one-sided, H1: pA<pB) = {p_one:.10f}")

# =========================
# 5) 결론
# =========================
alpha = 0.05
if p_one < alpha:
    print(f"결과: p(one-sided) < {alpha} 이므로 귀무가설(H0: pA=pB)을 기각합니다.")
    print("=> 검색바 상단 노출(B)이 상세 페이지 진입률(CVR)을 유의미하게 증가시킵니다. (H1: pA < pB 지지)")
else:
    print(f"결과: p(one-sided) ≥ {alpha} 이므로 귀무가설(H0)을 기각할 수 없습니다.")
    print("=> 상단 노출(B)의 CVR 증가 효과가 통계적으로 충분히 입증되지 않았습니다.")


pA_hat = 0.250000, pB_hat = 0.265000, (pB_hat - pA_hat) = 0.015000
Chi-square = 27.0668, p(two-sided) = 0.0000001965, p(one-sided, H1: pA<pB) = 0.0000000983
결과: p(one-sided) < 0.05 이므로 귀무가설(H0: pA=pB)을 기각합니다.
=> 검색바 상단 노출(B)이 상세 페이지 진입률(CVR)을 유의미하게 증가시킵니다. (H1: pA < pB 지지)


**결과해석**

답변:

1. 관측 결과, 대조군(A)의 상세 페이지 진입률(CVR): 25.0%, 실험군(B)의 상세 페이지 진입률(CVR): 26.5%로 실험군(B)이 대조군(A) 대비 더 높은 상세 페이지 진입률을 보였다.
2. 통계적 검정 결과, 카이제곱 통계량: χ² = 27.0668, 양측 p-value: 1.965 × 10⁻⁷, 단측 p-value (H₁: pA < pB): 9.83 × 10⁻⁸으로 관측된 CVR 차이가 H0가 참이라는 가정 하에 발생했을 가능성은 극히 낮다.

=> 따라서 검색바를 상단에 노출한 실험군의 상세 페이지 진입률은 대조군보다 유의미하게 높았으며(p < 0.001), 이에 따라 검색바 상단 노출은 사용자 전환을 개선하는 효과적인 UI 변경으로 판단된다.

<br>

**1주차 기초통계 과제는 여기까지였습니다. 수고하셨습니다!**